tests/integration/pre-build.tap.js

```diff
@@ -47,17 +72,42 @@ tap.test('pre-build commands', function (t) {
     t.end()
   })
 
-  // t.test('download', { skip: NO_PREBUILTS.includes(platform) }, function (t) {
-  // this will fail when we release because the new pre-built does not exist
-  t.test('download', { skip: true }, function (t) {
+  // no reason to test downloading on windows. only reason we test on mac is so
+  // you can run these tests locally
+  t.test('download', { skip: isWindows() }, async function (t) {
+    const downloadServer = fork(path.join(__dirname, './download-server.js'))
+    const port = await waitFor(downloadServer, 'STARTED')
+    process.env.NR_NATIVE_METRICS_DOWNLOAD_HOST = `http://localhost:${port}/`
     execSync('node ./lib/pre-build --no-build install native_metrics')
     const binary = findBinary()
     t.match(
       binary,
       /_newrelic_native_metrics-\d{1,3}_\d{1,3}_\d{1,3}-native_metrics.*\.node/,
       'should download binary'
     )
-    t.end()
+    downloadServer.kill()
+  })
+
+  // no reason to test downloading on windows. only reason we test on mac is so
+  // you can run these tests locally
+  t.test('download with proxy', { skip: isWindows() }, async function (t) {
+    const proxyServer = fork(path.join(__dirname, './proxy-server.js'))
+    const downloadServer = fork(path.join(__dirname, './download-server.js'))
+    const proxyPromise = waitFor(proxyServer, 'STARTED')
+    const downloadPromise = waitFor(downloadServer, 'STARTED')
+    const [proxyPort, downloadPort] = await Promise.all([proxyPromise, downloadPromise])
+
+    process.env.NR_NATIVE_METRICS_PROXY_HOST = `http://localhost:${proxyPort}`
+    process.env.NR_NATIVE_METRICS_DOWNLOAD_HOST = `http://localhost:${downloadPort}/`
+    execSync(`node ./lib/pre-build --no-build install native_metrics`)
+    const binary = findBinary()
+    t.match(
+      binary,
+      /_newrelic_native_metrics-\d{1,3}_\d{1,3}_\d{1,3}-native_metrics.*\.node/,
+      'should download binary'
+    )
+    proxyServer.kill()
+    downloadServer.kill()
   })
```

Comment:
Wonder if we could add a way to confirm we went through the proxy to the test.

Reply:
Hmm, not sure. I'll think on it